x/ccv/provider/keeper/grpc_query.go

```diff
@@ -333,17 +335,59 @@ func (k Keeper) QueryConsumerValidators(goCtx context.Context, req *types.QueryC
 
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	if _, found := k.GetConsumerClientId(ctx, consumerId); !found {
-		// chain has to have started; consumer client id is set for a chain during the chain's spawn time
-		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("no started consumer chain: %s", consumerId))
+	// check that the consumer is initialized
+	initParams, err := k.GetConsumerInitializationParameters(ctx, consumerId)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, errorsmod.Wrap(types.ErrUnknownConsumerId, consumerId).Error())
 	}
 
-	var validators []*types.QueryConsumerValidatorsValidator
+	// query consumer validator set
 
-	consumerValSet, err := k.GetConsumerValSet(ctx, consumerId)
-	if err != nil {
-		return nil, status.Error(codes.Internal, err.Error())
+	var consumerValSet []types.ConsensusValidator
+	// if the consumer hasn't launched yet, compute the consumer validator set
+	if ctx.BlockTime().Before(*initParams.SpawnTime) {
+		var bondedValidators []stakingtypes.Validator
+
+		powerParams, err := k.GetConsumerPowerShapingParameters(ctx, consumerId)
+		if err != nil {
+			return nil, status.Error(codes.InvalidArgument, types.ErrInvalidConsumerParams.Error())
+		}
+		bondedValidators, err = k.GetLastBondedValidators(ctx)
+		if err != nil {
+			panic(fmt.Errorf("failed to get last validators: %w", err))
```

Comment:
I understand that `panic` does not "hurt" here because it's a query but why not return an `error` instead?